pkg/projectfile/projectfile.go

```diff
@@ -102,6 +103,7 @@ type Project struct {
 	Cache         string        `yaml:"cache,omitempty"`
 	path          string        // "private"
 	parsedURL     projectURL    // parsed url data
+	commitID      string
```

Comment:
Commits are no longer a property of the projectile. And so there really shouldn't be any mention of commits in either the projectfile or the project packages other than to facilitate migrating.

For any commit logic that we keep around here for said migration I suggest we clearly prefix it as `deprecatedCommitID` so we don't accidentally use the wrong thing and it's clear to the reader that this isn't where the actual code lives.